### PR TITLE
amp-story utils: use shadow even in tests.

### PR DESCRIPTION
**Pull request by samouri**
Addresses https://github.com/ampproject/amphtml/pull/35889#issuecomment-910601025

### CODE REVIEW

File: build-system/tasks/visual-diff/index.js [hunk 1]
```diff
@@ -84,6 +84,12 @@ const REMOVE_NO_SCRIPT_ELEMENT_SNIPPET = fs.readFileSync(
   path.resolve(__dirname, 'snippets/remove-no-script.js'),
   'utf8'
 );
+
+const SERIALIZE_SHADOW = fs.readFileSync(
```
Comment (danielrozenberg) [inline build-system/tasks/visual-diff/index.js:88]: Supernit - `SERIALIZE_SHADOW_SNIPPET`, just cause the others have that suffix

File: build-system/tasks/visual-diff/snippets/serialize-shadow.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ */
+
+/**
+ * Serializes all open shadow roots on the page.
+ * @param {*} doc document to operate on.
+ */
+function serializeShadow(doc) {
+  doc.documentElement./*OK*/ innerHTML = doc.documentElement.getInnerHTML({
+    includeShadowRoots: true,
+  });
```
Comment (danielrozenberg) [inline build-system/tasks/visual-diff/snippets/serialize-shadow.js:15]: No way, it's that easy? 😅

File: extensions/amp-story-auto-ads/0.1/test/test-amp-story-auto-ads.js [hunk 3]
```diff
@@ -299,7 +301,9 @@ describes.realWin(
       });
 
       it('should create progress bar', () => {
-        const progressBar = doc.querySelector(
+        // throw new Error(doc.body.innerHTML);
```
Comment (danielrozenberg) [inline extensions/amp-story-auto-ads/0.1/test/test-amp-story-auto-ads.js:304]: Debugging leftover?

**Review by danielrozenberg — COMMENTED** [review]
Looks good! lmk if you need any help with the AIs you listed

File: extensions/amp-story-auto-ads/0.1/test/test-amp-story-auto-ads.js (hunk 3 above)
Comment (samouri) [reply extensions/amp-story-auto-ads/0.1/test/test-amp-story-auto-ads.js:304]: Still working on this file...it has a bunch of failing tests

File: build-system/tasks/visual-diff/snippets/serialize-shadow.js (hunk 2 above)
Comment (samouri) [reply build-system/tasks/visual-diff/snippets/serialize-shadow.js:15]: 😉

**Comment by samouri** [comment]
@danielrozenberg: FYI, my current plan is to make all the changes necessary to complete the amp-story migration within this PR, and then extract out the separable bits (like shadow dom serializatino)

**Pull request closed without merging**